tests/system/src/test/java/test/robot/javafx/scene/TransparentLCDTest.java

```diff
@@ -0,0 +1,273 @@
+        double a = c.getOpacity();
+        double r = c.getRed() * a + (1.0 - a);
+        double g = c.getGreen() * a + (1.0 - a);
+        double b = c.getBlue() * a + (1.0 - a);
+        return Color.color(r, g, b);
+    }
+
+    private boolean isGrayScale(List<Color> colors) {
+        long nonGrayCount = colors.stream()
+                .filter(c -> c.getRed() != c.getGreen() || c.getRed() != c.getBlue())
+                .count();
+        return nonGrayCount == 0;
+    }
+
+   private void assertColorEquals(Color expected, Color actual) {
+        if (!testColorEquals(expected, actual, TOLERANCE)) {
+            fail("expected:" + expected + " but was:" + actual);
+        }
+    }
+
+    private boolean testColorEquals(Color expected, Color actual, double delta) {
+        double deltaRed = Math.abs(expected.getRed() - actual.getRed());
+        double deltaGreen = Math.abs(expected.getGreen() - actual.getGreen());
+        double deltaBlue = Math.abs(expected.getBlue() - actual.getBlue());
+        double deltaOpacity = Math.abs(expected.getOpacity() - actual.getOpacity());
+        return (deltaRed <= delta && deltaGreen <= delta && deltaBlue <= delta && deltaOpacity <= delta);
+    }
+
+    private List<Color> getColors(Scene scene, int x, int y, int width) {
+        x += scene.getX() + scene.getWindow().getX();
+        y += scene.getY() + scene.getWindow().getY();
+        Image image = robot.getScreenCapture(null, x, y, width, 1);
+        List<Color> colors = new ArrayList<>(width);
+        for (int i = 0; i < width; i++) {
+            colors.add(image.getPixelReader().getColor(i, 0));
+        }
+        return colors;
+    }
+
+    // This must be called on the FX app thread
+    private Stage createStage() {
+        Stage stage = new Stage();
+        stage.initStyle(StageStyle.UNDECORATED);
+        stage.setAlwaysOnTop(true);
+        return stage;
+    }
+
+    @BeforeAll
+    public static void doSetupOnce() {
+        System.setProperty("prism.lcdText", "true");
+        System.setProperty("glass.win.uiScale", "1");
+        System.setProperty("glass.gtk.uiScale", "1");
+
+        Platform.setImplicitExit(false);
+        final CountDownLatch launchLatch = new CountDownLatch(1);
+        Util.startup(launchLatch, launchLatch::countDown);
+        assertEquals(0, launchLatch.getCount());
+    }
+
+    @AfterAll
+    public static void doTeardownOnce() {
+        Util.shutdown();
+    }
+
+    @BeforeEach
+    public void doSetup() {
+        // LCD text is disabled on macOS
+        assumeFalse(PlatformUtil.isMac());
+
+        // Test is not valid for SW pipeline. We don't have a utility to
+        // check the GraphicsPipeline, so we check for 3D support instead.
+        assumeTrue(Platform.isSupported(ConditionalFeature.SCENE3D));
+
+        Util.runAndWait(() -> robot = new Robot());
+        Util.parkCursor(robot);
+    }
+
+    @AfterEach
+    public void doTeardown() {
+        Platform.runLater(() -> {
```

Comment:
should this be Util.runAndWait() ?

Reply:
It doesn't really matter in this case since Runnables are run in order, meaning that the creation and showing of the stage for the next text is guaranteed to happen after the stage is hidden. I can change it, though.